crates/uv-requirements/src/specification.rs

```diff
@@ -1,10 +1,41 @@
-use std::collections::BTreeMap;
+//! Collecting the requirements to compile, sync or install.
```

Comment:
Should all of this logic instead be in `source_tree.rs`, like in the source tree resolver, instead of defined here?

Reply:
What parts do you mean? If it's a larger refactoring, i'd do it in another PR

Reply:
(Not for this PR, I suppose.)